software/database/src/main/java/org/apache/brooklyn/entity/database/mysql/MySqlNodeEffectors.java

```diff
@@ -75,7 +75,7 @@ public Void call(ConfigBag parameters) {
             DynamicTasks.queue(
                     SshTasks.newSshExecTaskFactory(machine, 
                                     "cd "+entity().getAttribute(MySqlNode.RUN_DIR),
-                                    "sed -i'' -e 's@^\\(\\s*password\\s*=\\s*\\).*$@\\1" + newPass.replace("\\", "\\\\") + "@g' mymysql.cnf")
+                                    "sed -i.bak 's/\\(password.*\\)=.*/\\1= " + newPass.replace("\\", "\\\\") + "/' mymysql.cnf")
```

Comment:
@tbouron Won't this replace lines containing the following as well?
```
passwordMine=xxx
passwordOld=yyy
passwordDontChange=zzz
```

What about `password=se!=scret`. I think it will match greedily here, so won't do the replacement right?

Is the change fixing the case for passwords containing `@`? Could you add some more details in the commit message.